src/main/java/net/neoforged/neoforge/fluids/FluidUnit.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.fluids;
+
+import com.mojang.serialization.Codec;
+import io.netty.buffer.ByteBuf;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.resources.ResourceLocation;
+
+public class FluidUnit {
+    private static final Map<ResourceLocation, FluidUnit> units = new ConcurrentHashMap<>();
+
+    public static Codec<FluidUnit> CODEC = ResourceLocation.CODEC.xmap(FluidUnit::get, FluidUnit::name);
+    public static StreamCodec<ByteBuf, FluidUnit> STREAM_CODEC = ResourceLocation.STREAM_CODEC.map(FluidUnit::get, FluidUnit::name);
+
+    ResourceLocation name;
```

Comment:
Is there a reason this is package-private, but the getter is private?

Reply:
Name being public would make life a lot easier, most notably in debug or serializing outside of codecs.

Reply:
Then he should create a getter for it.
The field should be fully private.

Reply:
made the getter public [also made the field final]